libraries/AP_DAL/LogStructure.h

```diff
@@ -0,0 +1,435 @@
+    uint8_t num_sensors;
+    uint8_t primary_sensor;
+    uint8_t _end;
+};
+
+// @LoggerMessage: RGPI
+// @Description: Replay Data GPS Instance
+struct log_RGPI {
+    uint32_t last_message_time_ms;
+    int32_t lat;
+    int32_t lng;
+    int32_t alt;
+    float hacc;
+    float vacc;
+    float lag_sec;
+    uint16_t hdop;
+    uint8_t status;
+    uint8_t have_vertical_velocity;
+    uint8_t horizontal_accuracy_returncode;
+    uint8_t vertical_accuracy_returncode;
+    uint8_t num_sats;
+    uint8_t get_lag_returncode;
+    uint8_t instance;
+    uint8_t _end;
+};
+
+// @LoggerMessage: RGPJ
+// @Description: Replay Data GPS Instance - more data
+struct log_RGPJ {
+    Vector3f velocity;
+    uint32_t speed_accuracy_returncode;
+    float sacc;
+    float yaw_deg;
+    float yaw_accuracy_deg;
+    uint8_t gps_yaw_deg_returncode;
+    uint8_t instance;
+    uint8_t _end;
+};
+
+// Replay Data Structure - Airspeed Sensor header
+struct log_RASH {
+    uint8_t num_sensors;
+    uint8_t primary;
+    uint8_t _end;
+};
+
+// Replay Data Structure - Airspeed Sensor instance
+struct log_RASI {
+    float airspeed;
+    uint32_t last_update_ms;
+    bool healthy;
+    bool use;
+    uint8_t instance;
+    uint8_t _end;
+};
+
+// @LoggerMessage: RMGH
+// @Description: Replay Data Magnetometer Header
+struct log_RMGH {
+    float declination;
+    uint8_t count;
+    bool auto_declination_enabled;
+    uint8_t num_enabled;
+    bool learn_offsets_enabled;
+    bool consistent;
+    uint8_t _end;
+};
+
+// @LoggerMessage: RMGI
+// @Description: Replay Data Magnetometer Instance
+struct log_RMGI {
+    uint32_t last_update_usec;
+    Vector3f offsets;
+    Vector3f field;
+    bool use_for_yaw;
+    bool healthy;
+    bool have_scale_factor;
+    uint8_t instance;
+    uint8_t _end;
+};
+
+// @LoggerMessage: RBCH
+// @Description: Replay Data Beacon Header
+struct log_RBCH {
+    Vector3f vehicle_position_ned;
+    float accuracy_estimate;
+    int32_t origin_lat;
+    int32_t origin_lng;
+    int32_t origin_alt;
+    bool get_vehicle_position_ned_returncode;
+    uint8_t count;
+    bool get_origin_returncode;
+    uint8_t ptr_is_nullptr;
+    uint8_t _end;
+};
+
+// @LoggerMessage: RBCI
+// @Description: Replay Data Beacon Instance
+struct log_RBCI {
+    uint32_t last_update_ms;
+    Vector3f position;
+    float distance;
+    uint8_t healthy;
+    uint8_t instance;
+    uint8_t _end;
+};
+
+// @LoggerMessage: RVOH
+// @Description: Replay Data Visual Odometry data
+struct log_RVOH {
+    Vector3f pos_offset;
+    uint32_t delay_ms;
+    uint8_t healthy;
+    bool enabled;
+    uint8_t ptr_is_nullptr;
+    uint8_t _end;
+};
+
+// @LoggerMessage: ROFH
+// @Description: Replay optical flow data
+struct log_ROFH {
+    Vector2f rawFlowRates;
+    Vector2f rawGyroRates;
+    uint32_t msecFlowMeas;
+    Vector3f posOffset;
+    uint8_t rawFlowQuality;
+    uint8_t _end;
+};
+
+// @LoggerMessage: REPH
+// @Description: Replay external position data
```

Comment:
```suggestion
// @Description: Replay external position header
```

Reply:
it's data (no following instance). name to match patterns